Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ApplePayTokenizationViewModel.swift

```diff
@@ -303,30 +345,80 @@ class ApplePayTokenizationViewModel: PaymentMethodTokenizationViewModel {
             }
         }
     }
+
+    func getShippingMethodsInfo() -> ShippingMethodsInfo {
+        guard let options = PrimerAPIConfigurationModule
+            .apiConfiguration?
+            .checkoutModules?
+            .first(where: { $0.type == "SHIPPING"})?
+            .options as? ShippingMethodOptions else {
+            return .init(shippingMethods: nil, selectedShippingMethodOrderItem: nil)
+        }
+
+        var factor: Double
+        if AppState.current.currency?.isZeroDecimal == true {
+            factor = 1
+        } else {
+            factor = 100
+        }
+
+        // Convert to PKShippingMethods
+        let apShippingMethods = options.shippingMethods.map {
+            let amount: Double = Double($0.amount)/factor
```

Comment:
I'm not sure what the margin of error on precision is here but could do one of a couple of things if it's relevant:

1. round to a fixed two digits (not very precise)
2. perform the operation on the NSDecimalNumber to avoid precision issues (using `dividing(by:)`)

(I say "if it's relevant" because maybe PassKit handles this already, I'm not sure)

Reply:
Nice, thanks for this. Will update